src/test/util.ts

```diff
@@ -144,4 +144,13 @@ export function sanitizeForFilePath(s: string) {
   return s.replace(/[\x00-\x2F\x3A-\x40\x5B-\x60\x7B-\x7F]+/g, '-');
 }
 
+/**
+ * Get absolute path contained within parent directory, otherwise returns parent path
+ */
+export function getContainedPath(parentPath: string, subPath?: string): string {
+  if (!subPath) return parentPath;
+  const resolvedPath = path.resolve(parentPath, subPath);
+  return resolvedPath.startsWith(parentPath) ? resolvedPath : path.join(parentPath, path.basename(subPath));
```

Comment:
I think this needs to be `startsWith(parentPath  + path.sep)`, otherwise `toMatchSnapshot('../foo.spec.js-snapshots123/bar.txt')` will escape.

Reply:
I don't exactly follow, are saying in the event someone passes a path the goes to the parent then back into the correct snapshot directory it will not escape the path? If I were to test something like...

```ts
test.only('should prevent reassigning snapshot test path', async ({ runInlineTest }) => {
  const result = await runInlineTest({
    ...files,
    'a.spec.js-snapshots/snapshot.txt': `Hello world`,
    'a.spec.js': `
      const { test } = require('./helper');
      test('is a test', ({}) => {
        expect('Hello world').toMatchSnapshot(['..', 'a.spec.js-snapshots', 'snapshot.txt']);
      });
    `
  });

  expect(result.exitCode).toBe(0);
});
```

This test fails because `outputPath` path is expecting `/test-results/a-is-a-test` and not `/test-results/a.spec.js-snapshots` since the `outputPath` is formatted differently than snapshots.

```
  1) a.spec.js:6:7 › is a test =====================================================================

    Error: Path defined outside of parent directory
      path: ../a.spec.js-snapshots/snapshot.txt
      resolved path: /Users/nicholaspartridge/Documents/repos/playwright/test-results/golden-should-prevent-reassigning-snapshot-test-path-playwright-test/test-results/a.spec.js-snapshots/snapshot.txt
      parent: /Users/nicholaspartridge/Documents/repos/playwright/test-results/golden-should-prevent-reassigning-snapshot-test-path-playwright-test/test-results/a-is-a-test

      5 |       const { test } = require('./helper');
      6 |       test('is a test', ({}) => {
    > 7 |         expect('Hello world').toMatchSnapshot(['..', 'a.spec.js-snapshots', 'snapshot.txt']);
        |                               ^
      8 |       });
```

This seems like an edge case that could be resolved with the new error message. Is your intent for this case to throw on `snapshotPath`?